### DISCUSSION TITLE
How to use custom NAPALM-drivers?

### DISCUSSION THREAD

**Discussion by Johannes0317** [nautobot/nautobot#3989] [Q&A] [answered]
The driver I want to use is: https://github.com/napalm-automation-community/napalm-arubaos-switch
I have installed it, but now I´m struggling with mapping the slug for the napalm driver to the correct napalm driver.
How can I achieve this?

**Comment by joewesch** [comment]
Do you have `arubaoss` set on the Platform as the NAPALM driver?

**Reply by Johannes0317** [reply]
Yes, but now when I'm running e.g. Golden Config Task Backup Config I get this error: "There was no username (redacted) preemptively failed."
Where do I have to set the username? I tried setting the NAPALM_USERNAME in the nautobot config. I also tried setting a name in the plugins config of nautobot-nornir.
Furthermore I tried adding the username to the secrets-group of the device.

**Reply by itdependsnetworks** [reply]
This is for https://github.com/nautobot/nautobot-plugin-golden-config or https://github.com/nautobot/nautobot-plugin-nornir, but probably best to join Slack for help.

**Reply by Johannes0317** [reply, marked as answer]
AH ok, I will seek help on the network to code slack. Thanks for your help.

**Reply by syukrighazali95** [reply]
Hi, @Johannes0317, may I know if you have the solution for this?  I am also facing the same issue even though I have defined the NAPALM_USERNAME in environment variables. Really appreciate your help. Thank you
